---
title: "KI Platygyra Manuscript"
author: "Danielle C Claar, Kristina L Tietjen, Ruth D Gates, Julia K Baum"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    number_sections: yes
  html_document:
    fig_caption: yes
    force_captions: yes
    highlight: pygments
    number_sections: yes
    theme: cerulean
csl: nature.csl
bibliography: PapersLibrary_Bibtex.bib
---

<!-- ## Install and source necessary packages -->
```{r, echo=FALSE, incl=FALSE}

  # install.packages('knitcitations')
  # install.packages('bibtex')
  # install.packages('qdap')

  # devtools::install_github("cboettig/knitcitations@v1")
  library(knitcitations); cleanbib()
  cite_options(citation_format = "pandoc", check.entries=FALSE)
  library(bibtex)
  
  # source('http://bioconductor.org/biocLite.R')
  # biocLite('phyloseq')
  # packageVersion('phyloseq')
  library(phyloseq)
  library("qdap") # to be able to use word count
```

<!-- Word Count -->
```{r, echo=FALSE, warning=FALSE, message=FALSE, incl=FALSE}
  wc("") # to count words
```
*Titles should not exceed 75 characters (including spaces) for Articles. Titles should not include numbers, acronyms, abbreviations or punctuation. They should include sufficient detail for indexing purposes but be general enough for readers outside the field to appreciate what the paper is about.*
  
*Letters no more than 4 pages, of Nature. An uninterrupted page of text contains about 1,300 words.*
  
  
*Letters are short reports of original research focused on an outstanding finding whose importance means that it will be of interest to scientists in other fields. As a guideline they allow up to 30 references. They begin with a fully referenced paragraph, of about 200 words, (certainly no more than 300 words) aimed at readers in other disciplines. This starts with a 2–3 sentence basic introduction to the field; followed by a one-sentence statement of the main conclusions starting ‘Here we show’ or equivalent phrase; and 2–3 sentences putting the main findings into general context. See the information sheet 'How to construct a Nature summary paragraph' for an annotated example. The rest of the text is typically about 1,500 words long (not including Methods, summary paragraph or other sections). Letters have 3 or 4 small display items.*

# Summary
Coral reefs, which already live on the edge of their thermal tolerance (CITE), are under acute threat from El Nino-associated ocean warming (CITE). The 2015/16 El Nino is currently the worst on record in terms of severity and longevity (@Eakin:2016vf), yet despite massive coral mortality, some corals show resilience to this extreme event. El Nino warming is intensifying (@Trenberth:1997vw, @Cobb:2013fe, @Cai:2014ky), threatening coral reefs and endangering the persistence of vital ecosystem services, threatening the food security and coastline protection of coastal communities worldwide (CITE). Coral resilience is related to many factors (CITE), including the structure and flexibility of their internal symbiotic communities (CITE,CITE). However, the mechanisms of coral resilience to extended pulse stress events are still poorly understood. Here, using Illumina MiSeq amplicon sequencing, we show that, contrary to current opinion (CITE), symbionts present in miniscule abundances (<2%) are indeed important for coral recovery. *Additionally, we show that the ability of a coral to house a diverse suite of symbionts (*Symbiodinium*) is driven by the identity of the dominant *Symbiodinium* type. *This isn't necessarily true!* Furthermore, we show that corals have the ability to regain symbionts during an extended stress event, providing hope for the future of coral resilience.

# Main Text
```{r, echo=FALSE, include=FALSE}
citep("Article")
```
*The rest of the text is typically about 1,500 words long (not including Methods, summary paragraph or other sections). Letters have 3 or 4 small display items.*

*Little bit more background*
Ocean ecosystems worldwide are threatened by climate change-induced increases in seawater temperatures (CITE, CITE, CITE). Pulse warming events such as El Nino amplify these threats, causing massive losses of coral cover (e.g. 17% of coral reefs during the 1997/98 El Nino) (CITE,CITE). El Niño, the positive phase of the ENSO (El Niño Southern Oscillation), is a natural climatic event that occurs when surface waters heat up in the equatorial Pacific, that causes catastrophic effects on reef ecosystems by disrupting coral symbioses (i.e. coral bleaching). El nino warming threatens coral reefs by disrupting the dynamic symbiosis between coral and their internal symbiotic algae (Symbiodinium). This symbiosis is the foundation of reef ecosystems, and a critical element of reef resilience (van oppen and gates 2006). Corals host a diverse community of Symbiodinium, ranging along a continuum from 'selfish opportunistic symbionts' (e.g. some clade D Symbiodinium) which are better suited to sustained environmental stress than others, to 'intimately evolved symbionts' which provide exceptional amounts of nutrition to their coral host (@Lesser:2013tu). Thus, although these relationships have developed over evolutionary time, the resilience of the coral symbiome is constantly shaped by dynamic coral-symbiont interactions (@Stat:2006ww). The 2015-2016 El Niño is the first major global event since 1997-1998, and has been declared the third global coral bleaching event by NOAA (@Eakin:2016vf).  Kiritimati Atoll (Christmas Island), located in the Central Equatorial Pacific, is at the epicenter of this extreme El Niño event. Thermal anomalies were severe on Kiritimati, reaching an unprecedented (cite o h-g) XX number of DHW over a XX month long bleaching event, demolishing most of the reef (@Baum_inprep). 

*Hough-Guldberg 2011* - good segment on DHW and bleaching (excellent figure for bleaching..never before seen on reefs...)

*Briefest methods plus results summary*
Here we assess *briefest methods section ever* Here, we tagged and sampled the same corals before, during, and immediately after the el nino event, on Kiritimati (something about Kiritimati). We used Illumina sequencing to evaluate changes in symbiodinium community structure coincident with the 2015-2016 major el nino event. The goal was to understand...why the hell these corals survived 10 months of extreme heat stress, and actually got better in the middle of it.
Example of when this high-risk ecological opportunity (@Baker:2001bf) actually pays off...

# *Tiny abundances are important*
 *We used to think that bleaching might be good - ABH says that corals bleach in order to expel suboptimal Symbiodinium types in exchange for optimal symbionts during the new conditions (@Buddemeier:1993bb, Baker:2001bf, @Buddemeier:2004vj)* *We do know that corals house background symbionts in low abundances (@Correa:2009jy, all the recent ngs studies...), but these relationships have been described as unstable (@Coffroth:2010ju, more cites?). *Switching and shuffling (@Baker:2003uo)* *And we do know that some symbio are "better" than others* *And then we said that bleaching is definitely bad* *But at least we do know that it allows changes to occur in the Symbiodinium community structure*
*Figure of example sequence abundances, superimposed on coral 99 images*

# *Dominant symbiont drives sym div*
*Regardless it is likely that symbiodinium community composition is important for resilience, corals that host flexible symbioses may be more sensitive to environmental changes (@Putnam:2012bn)* *Taxa-specific bleaching is a thing* 
*Figure about diversity*

# *Corals can regain symbionts, whoa!*
*When we've seen resilience/recovery before, corals have only been demonstrated to recover if the stress goes away first* Previously, corals have been shown to recover from bleaching only after the external stress (e.g. warming) has subsided. *implying that longer and more frequent stressors spell disaster for reefs worldwide*

# *In a sea of destruction, a glimmer of hope*
*we have shown....why this is important*
(Methods like NGS are really important for understanding these changes in symbiont diversity, as well as for seeing those low aboundance symbionts)
(What does their recovery tell us about the future of coral reefs?)... Why are Platys so excellent and what does that tell us about when coral resilience is threatened by extreme climatic events?
Elucidating the mechanisms underlying changes in coral-symbiont interactions is essential to understanding the ability of the coral symbiome to adapt to the multiple stressors they now face. 

# Methods
*The Methods section should be written as concisely as possible but should contain all elements necessary to allow interpretation and replication of the results. As a guideline, Methods sections typically do not exceed 3,000 words. Detailed descriptions of methods already published should be avoided; a reference number can be provided to save space, with any new addition or variation stated. The Methods section should be subdivided by short bold headings referring to methods used and we encourage the inclusion of specific subsections for statistics, reagents and animal models. If further references are included in this section, the numbering should continue from the end of the last reference number in the rest of the paper and the list should accompany the additional Methods at the end of the paper. The Methods section cannot contain figures or tables (essential display items should be included in the Extended Data).*

``````{r, echo=FALSE, warning=FALSE, message=FALSE, eval=FALSE}
path <- "/Volumes/Seagate_Blue/JuliaBaum_DanielleClaar-32138116_all/"
biom_dir <- "otu_tables/"
biom_file <- paste(path, biom_dir, "KI14FSYM004_otu_table.biom", sep = "")
map_dir <- 
map_file <- paste(path, map_dir, "", sep = "")
ref_dir <- 
ref_file <- 

biom_otu_tax <- import_biom(biom_file, "greengenes")


# Add sample data to the dataset using merge
bmsd <- import_qiime_sample_data(map_file)
class(bmsd)
Bushman <- merge_phyloseq(biom_otu_tax, bmsd)

import_biom(biom, treefilename, refseqfilename, parseFunction=parse_taxonomy_greengenes)
import_biom(rich_dense_biom, treefilename, refseqfilename, parseFunction=parse_taxonomy_greengenes)

```


## Field Information
*Kiritimati basics - located in the Central Equatorial Pacific, smack dab in the middle of the Nino 3.4 region (used to quantify el nino presence and strength), human disturbance gradient, bleaching event there (cite bleaching paper here)*
*Tagging corals and collecting samples - transects, tagging corals, photoing corals, sampling corals, processing samples, storage in Guanidinium*
*Taxa sampled - platy, favites, favia, etc.*

Kiritimati Atoll, Kiribati is located in the Central Equatorial Pacific and has a strong human disturbance gradient around the island. The atoll was in the middle of the El Niño 3.4 region (used to quantify El Niño presence and strength) during the 2015/2016 El Niño and experienced 10 months of elevated temperature causing a mass bleaching and mortality event (cite bleaching paper). Temperature loggers (Seabird 56) were deployed around the island at 10-12m depths starting in summer 2013 to measure the temperature *in situ*.

##Coral Tagging and sampling  
In August/September 2014 seven taxa of coral were tagged along a 60m transect at 10-12m depth at 10 different sites around the atoll.  At each site, 12 colonies of each focal species (*Porites lobata*, *Montipora aequituberculata*, and *Pocillopora eyedouxi*) were tagged and a total of 18 colonies of a varying assortment of *Platygyra*, *Favities*, *Favia*, and *Hydnophora microconos* were tagged. A photo of each coral with a ruler beside it was taken for colony measurments and characterists (i.e. bleaching) to be analyzed in the lab. A tissue sample was collected and put into a Whirl-pak for transport.  The tissue samples were stored in a cooler on the boat and at night they were rinsed with fresh water and alliquoted into centrifuge tubes with Guanidinium, then stored at 4 (I havent looked to see how to put in a degree symbol yet) C.  The tagged coral colonies were resampled once more before the El Niño (January/February 2015), twice during (May/April 2015 and July 2015), and once near the end (March 2016).  Due to weather conditions not every site was visited on each trip. Two sites were added in January/February 2015 and one in May/April 2015.  Due to either storm waves breaking and moving coral or the tags dissapearing some of the corals were unable to be located and new corals were tagged to replenish numbers **I(KT) dont like this sentence - still working on how to word it*.


## Pre-processing and sequencing
Extraction of the nucleic acids from the tissue samples were performed as in Stat et al. (2009) (*need to add to references and format correctly*) with a few modifications. The DNA pellet was washed with 70% ethanol three times.  After the wash was removed, the remaining ethanol was allowed to evaporate off before the DNA pellet was resuspended in the Tris buffer.

DNA Extraction - extraction protocol
ITS2 region - it's annoying, but it's the best we've got right now
PCR and Cleanup - Amy's method of PCR and cleanup
Library Prep - Amy's method of library prep, include Illumina Sequencing information (barcodes, etc)

## Post Processing - Maybe call it Bioinformatics
Sequence QC - boku, then merge with illumina utils, max mismatch=3
Sequence clustering - denovo clustering using UCLUST in QIIME, then compare to reference database to assign taxonomy
Statistical Analysis - alpha diversity of sequence reads, co-occurence?, beta diversity?
Code will be avaible on git hub

# References
<!-- Insert references  -->
```{r, warning=FALSE, message=FALSE, echo=FALSE} 
  write.bibtex(entry=NULL, file="knitcitations.bib", append=FALSE)
```
